src/utilities/aggregateLogs.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+/**
+ * Aggregates sensor logs into groups of logs by dates.
+ *
+ * @param   {Realm.results}  sensorLogs          A collection of sensorLog objects to aggregate.
+ * @param   {boolean}        isMax               If true, aggregate by maximum temperature, else
+ *                                               by minimum temperature.
+ * @param   {number}         numberOfDataPoints  Number of aggregated data points to return.
+ * @param   {Date}           startDate           Start date of temperature range to aggregate
+ *                                               over.
+ * @param   {Date}           endDate             End date of temperature range to aggregate over.
+ * @return  {Object[]}                           Aggregated sensor logs, array of objects in form
+ *                                              [{timestamp}, {temperature},...].
+ */
+export const aggregateLogs = ({
+  sensorLogs,
+  numberOfIntervals,
+  isMax = true,
+  startDate = new Date(),
+  endDate = new Date(),
+}) => {
+  // Break out if no sensorLogs.
+  if (!(sensorLogs.length > 0)) return [];
+
+  // Sort sensor logs ascending by date.
+  sensorLogs.sorted('timestamp');
+
+  // If start date is later than earliest log, update.
+  const startLog = sensorLogs[0];
+  startDate = startDate > startLog.timestamp ? startLog.timestamp : startDate;
```

Comment:
This is a downside of passing objects as parameters. Assigning new values to parameters causes more side effects than usual. 

In any case it's usually discouraged to reassign any value for the usual reasons and apparently most JS engines are not optimized for reassignment. Also I've read some say it is worse to reassign parameters (as this can mutate the `arguments` object, but I think this argument isn't very valid as it's also discouraged to access the `arguments` object).

Reply:
I didn't like reassigning either, I was actually surprised the args weren't constants. Fixed by 6f9f29dd.